src/sentry/static/sentry/app/views/settings/incidentRules/metricField.tsx

```diff
@@ -0,0 +1,200 @@
+import React from 'react';
+import styled from '@emotion/styled';
+import {css} from '@emotion/core';
+
+import FormField from 'app/views/settings/components/forms/formField';
+import {t, tct} from 'app/locale';
+import {QueryField} from 'app/views/eventsV2/table/queryField';
+import {generateFieldOptions} from 'app/views/eventsV2/utils';
+import {FieldValueKind} from 'app/views/eventsV2/table/types';
+import {Organization} from 'app/types';
+import space from 'app/styles/space';
+import FormModel from 'app/views/settings/components/forms/model';
+import Button from 'app/components/button';
+import Tooltip from 'app/components/tooltip';
+import {
+  explodeFieldString,
+  generateFieldAsString,
+  AggregationKey,
+  FieldKey,
+  AGGREGATIONS,
+  FIELDS,
+} from 'app/utils/discover/fields';
+
+import {Dataset} from './types';
+
+type Props = Omit<FormField['props'], 'children' | 'help'> & {
+  organization: Organization;
+};
+
+const cannedAggregates = [
+  {
+    match: /^count\(\)/,
+    name: 'Number of errors',
+    validDataset: [Dataset.ERRORS],
+    default: 'count()',
+  },
+  {
+    match: /^count_unique\(user(\.id)?\)/,
+    name: 'Users affected',
+    validDataset: [Dataset.ERRORS],
+    default: 'count_unique(user)',
+  },
+  {
+    match: /^(p[0-9]{2}|percentile\(transaction\.duration,[^)]+\))/,
+    name: 'Latency',
+    validDataset: [Dataset.TRANSACTIONS],
+    default: 'percentile(transaction.duration, 0.95)',
+  },
+  {
+    match: /^apdex\([0-9.]+\)/,
+    name: 'Apdex',
+    validDataset: [Dataset.TRANSACTIONS],
+    default: 'apdex(300)',
+  },
+  {
+    match: /^count\(\)/,
+    name: 'Throughput',
+    validDataset: [Dataset.TRANSACTIONS],
+    default: 'count()',
+  },
+  {
+    match: /^error_rate\(\)/,
+    name: 'Error rate',
+    validDataset: [Dataset.TRANSACTIONS],
+    default: 'error_rate()',
+  },
+];
+
+type OptionConfig = {
+  aggregations: AggregationKey[];
+  fields: FieldKey[];
+};
+
+const errorFieldConfig: OptionConfig = {
+  aggregations: ['count', 'count_unique'],
+  fields: ['user'],
+};
+
+const transactionFieldConfig: OptionConfig = {
+  aggregations: [
+    'avg',
+    'percentile',
+    'error_rate',
+    'apdex',
+    'count',
+    'p50',
+    'p75',
+    'p95',
+    'p99',
+    'p100',
+  ],
+  fields: ['transaction.duration'],
+};
+
+const getFieldOptionConfig = (dataset: Dataset) => {
+  const config = dataset === Dataset.ERRORS ? errorFieldConfig : transactionFieldConfig;
+
+  const aggregations = Object.fromEntries(
+    config.aggregations.map(key => [key, AGGREGATIONS[key]])
+  );
+  const fields = Object.fromEntries(config.fields.map(key => [key, FIELDS[key]]));
+
+  return {aggregations, fields};
+};
+
+const help = ({name, model}: {name: string; model: FormModel}) => {
+  const aggregate = model.getValue(name) as string;
+
+  const presets = cannedAggregates
+    .filter(preset => preset.validDataset.includes(model.getValue('dataset') as Dataset))
```

Comment:
Maybe we should document that this field needs to be in a form with a `dataset` property for it to do useful switching.